selectors.go

```diff
@@ -32,153 +29,215 @@ const (
+	for k, v := range selectorToChainDetails {
+		copyMap[k] = v
 	}
 
-	var data ymlData
-	err := yaml.Unmarshal(ymlFile, &data)
-	if err != nil {
-		panic(err)
+	return copyMap
+}
+
+func TestChainSelectorToChainDetails() map[uint64]ChainDetails {
+	copyMap := make(map[uint64]ChainDetails, len(selectorToChainDetails))
+	for k, v := range testSelectorToChainDetailsMap {
+		copyMap[k] = v
 	}
 
-	return data.Selectors
+	return copyMap
 }
 
 func GetSelectorFamily(selector uint64) (string, error) {
-	family, exist := selectorToChainFamily[selector]
-	if !exist {
-		return "", fmt.Errorf("family not found for selector %d", selector)
+	// previously selector_families.yml includes both real and test chains, therefore we check both maps
+	details, exist := selectorToChainDetails[selector]
+	if exist {
+		return details.Family, nil
+
 	}
 
-	return family, nil
+	details, exist = testSelectorToChainDetailsMap[selector]
+	if exist {
+		return details.Family, nil
+	}
+
+	return "", fmt.Errorf("chain detail not found for selector %d", selector)
 }
 
-func EvmChainIdToChainSelector() map[uint64]uint64 {
-	copyMap := make(map[uint64]uint64, len(evmChainIdToChainSelector))
-	for k, v := range evmChainIdToChainSelector {
-		copyMap[k] = v.ChainSelector
+func ChainIdFromSelector(chainSelectorId uint64) (string, error) {
+	chainDetail, ok := selectorToChainDetails[chainSelectorId]
+	if ok {
+		return chainDetail.ChainID, nil
 	}
-	return copyMap
-}
 
-func ChainIdFromSelector(chainSelectorId uint64) (uint64, error) {
-	for k, v := range evmChainIdToChainSelector {
-		if v.ChainSelector == chainSelectorId {
-			return k, nil
-		}
+	chainDetail, ok = testSelectorToChainDetailsMap[chainSelectorId]
+	if ok {
+		return chainDetail.ChainID, nil
 	}
-	return 0, fmt.Errorf("chain not found for chain selector %d", chainSelectorId)
+
+	return "0", fmt.Errorf("chain not found for chain selector %d", chainSelectorId)
 }
 
-func SelectorFromChainId(chainId uint64) (uint64, error) {
-	if chainSelectorId, exist := evmChainIdToChainSelector[chainId]; exist {
-		return chainSelectorId.ChainSelector, nil
+func SelectorFromChainIdAndFamily(chainId string, family string) (uint64, error) {
+	// if family is missing use EVM as default
+	if family == "" {
+		family = FamilyEVM
+	}
+
+	selectorMap, exist := chainIDToSelectorMapForFamily[family]
+	if !exist {
+		return 0, fmt.Errorf("chain selector map not found for family %v", family)
+	}
+
+	selector, exist := selectorMap[chainId]
+	if !exist {
+		return 0, fmt.Errorf("chain selector not found for chainID %v, family %v", chainId, family)
 	}
-	return 0, fmt.Errorf("chain selector not found for chain %d", chainId)
+
+	return selector, nil
 }
 
-func NameFromChainId(chainId uint64) (string, error) {
-	details, exist := evmChainIdToChainSelector[chainId]
+func NameFromChainIdAndFamily(chainId string, family string) (string, error) {
+	// if family is missing use EVM as default
```

Comment:
Instead of duplicating all this code from above function, just call that?
SelectorFromChainIdAndFamily()
Then from selector, just fetch details.